### PR TITLE
Introduce native AssemblyLoadContext base class

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/src/vm/assemblyloadcontext.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// See the LICENSE file in the project root for more information.
+
+#ifndef _ASSEMBLYLOADCONTEXT_H
+#define _ASSEMBLYLOADCONTEXT_H
+
+//
+// Unmanaged counter-part of System.Runtime.Loader.AssemblyLoadContext
+//
+class AssemblyLoadContext : public IUnknownCommon<ICLRPrivBinder, IID_ICLRPrivBinder>
+{
```
Comment (jkotas) [inline src/coreclr/src/vm/assemblyloadcontext.h]: These is a lot more that can move here.

**Review by MichalStrehovsky — APPROVED** [review]
Looks good to me, but I'm no expert in this codebase. Thank you!

**Review by vitek-karas — APPROVED** [review]
Looks good

**Review by elinor-fung — APPROVED** [review]
I'm actually unclear how we decide (decided?) where binder-related code should live (even the three binder implementations are split between binder and vm right now), so my only question would be why the new `AssemblyLoadContext` base should go under vm versus binder.

**Comment by jkotas** [comment]
I think we should work on making binder directory, ICLRPrivBinder, etc. to go away. Ideally, the low-level details like `AssemblyLoadContext` unloading would stay in VM, but the rest of it would be in C#.

**Comment by jkotas** [comment]
Failures are known issue #32126

**Pull request merged**